smartsheet/models/event.py

```diff
@@ -46,6 +46,12 @@ def __init__(self, props=None, base_obj=None):
         if props:
             deserialize(self, props)
 
+            # If additionalDetails is expected to be a complex object and needs to be fully loaded
+            if 'additionalDetails' in props:
+                # Assuming additionalDetails is a dictionary that needs to be fully loaded
+                # At this point additional details has only one level of nesting
+                self._additional_details = props['additionalDetails']
```

Comment:
A more pythonic way to do this perhaps would be...
```py
            self._additional_details = props.get('additionalDetails', None)
```
That of course assumes having `self._additional_details set to `None` is acceptable, vs the method your are suggesting here which leaves it completely unset. The preferred method would be to have it set to None to prevent potential downstream KeyError exceptions.